content/en/blog/2021/content/content.md

```diff
@@ -0,0 +1,65 @@
+---
+title: 
+spelling: 
+author:  
+date: 2021-08-17
+draft: false
+---
+Please read the article by [Article by Martin](https://martin.kleppmann.com/2016/02/08/how-to-do-distributed-locking.html) before going to this blog.
+Why can't lease based lock service provide a mutual exclusion to processes?
+Reason : This is because such a lease mechanism depends on the physical clock of both the lock service and client processes. Many factors.
+
+Now the question arises, how can we solve this problem? Such a problem can be solved with a technique called version number validation or fencing tokens. With this technique a shared resource needs to validate requests from clients based on their tokens.
+
+What is Version Number Validation or fencing tokens?
+When using a lock or lease to protect access to some resource, such as the file storage in Figure 8-4, we need to ensure that a node that is under a false belief of being “the chosen one” cannot disrupt the rest of the system.
+
+The directory contains two programs: 
+1. client. 
+2. storage.
+
+With etcd, you can reproduce the expired lease problem of distributed locking and a simple example solution of the validation technique which can avoid incorrect access from a client with an expired lease.
+
+How does it work?
+1. storage : storage works as a very simple key value in-memory store which is accessible through HTTP and a custom JSON protocol.
+2. client : client works as client processes which tries to write a key/value to storage with coordination of etcd locking.
+
+How to build them?
+In order to  build client and storage, just execute go build in each directory.
+
+How to try them?
+1. At first you need to start an etcd cluster
+2. On top of the etcd source directory, execute commands like:
+- $ ./build      # build etcd
+$ goreman start

```

Comment:
I'd suggest using a code fence rather than a bullet for code/terminal samples like this.

```
$ ./build      # build etcd
$ goreman start
```

(a code fence is surrounding a bit of  code with "```" on the lines above and below it)